docs/rfcs/logging-before-config-resolution.md

```diff
@@ -0,0 +1,64 @@
+# How To Log Before Config Resolution
+
+## Overview
+
+The OpenTelemetry Collector supports configuring a primary logger that the collector and its components use to write logs.
+This logger cannot be created until the user's configuration has been completely resolved.
+There is a need to write logs during the collector start-up, before the primary logger is instantiated, such as during
+configuration resolution or config validation. This document describes
+
+- why providing logging capabilities during startup is important
+- the current (as of v0.99.0) behavior of the Collector
+- different solutions to the problem
+- the accepted solution
+
+## Why Logging During Startup is Important
+
+When the collector is starting it tries to resolve user configuration as quickly as possible.
+But the Collector's configuration resolution strategy is not trivial - it allows for complex interactions between
+multiple, different config sources that must all resolve without error. During this process important information could
+be shared with users such as:
+- [Warnings about deprecated syntax](https://github.com/open-telemetry/opentelemetry-collector/issues/9162)
+- [Warnings about undesired, but handled, situations](https://github.com/open-telemetry/opentelemetry-collector/issues/5615)
+- Debug information
+
+## Requirements for any solution
+
+1. Once the primary logger is instantiated, it should be usable anywhere in the Collector that does logging.
+2. Log timestamps must be accurate to when the log was written, regardless of when the log is written to the user-specified location(s)
+3. If an error occurs before the primary logger is created any previously written logs MUST be written to the either stout or stderr if they have not already been written there.
```

Comment:
@mx-psi I added back the `stdout`/`stderr` requirement since it is not possible to write to a user-specific location without the primary logger, and in this situation the collector is terminating and primary logger could not be created.

Reply:
Right, makes sense, the change was only needed in (2)